decidim/decidim
# Add official blog posts

#### :tophat: What? Why?

This PR implements the final details for #8793, so for the full explanation you should go to that PR. As a summary: 

> Blog posts can have the organization as author.

What I personally added here: 

1. Merged with develop, as it has stagnated for a couple of months.  
2. Fix the bug detected by @ahukkanen with the API [at the review](https://github.com/decidim/decidim/pull/8793#pullrequestreview-972225631). I implemented what we already do with `Proposals`/`Coauthorable`: official resources return nil. 
3. In the meantime, I also found that we hadn't updated the `MeetingType` wasn't consistent with the other resources (like `DebateType`), as it was implementing `AuthorInterface` instead of `AuthorableInterface`, so I've also fixed that too. 

#### :pushpin: Related Issues

- Fixes #7500

#### Testing

0. Run the seeds in a clean database. See that you have "Official post" 
1. Go to a blog component in a process
4. Create a post
5. See that you can add the organization as the author. Save it 
6. Go to the frontend. See that it's official, like Meetings/Proposals/Debates 
7. Go to the API at `/api/graphiql`. Find a blog component ID, change 19 for that one. Execute: 

```graphql
{
  component(id: 19) {
    ... on Blogs {
      posts {
        nodes {
          id
          title {
            translations {
              locale
              text
            }
          }
          author {
            name
            avatarUrl
          }
        }
      }
    }
  }

```
:hearts: Thank you!
